examples/cmd/nse/nse_run.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+kill_nse() {
+    kill -s "$1" "${NSE}"
+}
+
+sighup () {
+    kill_nse "HUP"
+}
+
+sigint () {
+    kill_nse "INT"
+}
+
+sigterm () {
+    kill_nse "TERM"
+}
+
+sigquit () {
+    kill_nse "QUIT"
+}
+
+trap sighup HUP
+trap sigint INT
+trap sigterm TERM
+trap sigquit QUIT
+
+echo Starting NSE: "${NSE_IMAGE}"
+"/bin/${NSE_IMAGE}" &
+
+NSE=$!
+wait "${NSE}"
+wait "${NSE}"
```

Comment:
Why 2 times `wait` here?

Reply:
When the container stops, the shell process receives the signal, breaks `wait` execution, sends the received signal to the NSE process with the `kill` and then it should wait for the NSE process to complete all shutdown hooks.
If we omit this second wait, the NSE process will also receive the signal but the container would be destroyed before the shutdown hooks got executed.